components/proxy/src/main/java/com/hotels/styx/ServerConfigSchema.java

```diff
@@ -30,133 +31,130 @@
+                                    field("encoding", object(
+                                            field("unwiseCharactersToEncode", string())
+                                    ))
+                            )),
+                            optional("request-logging", object(
+                                    optional("inbound", logFormatSchema),
+                                    optional("outbound", logFormatSchema),
+                                    atLeastOne("inbound", "outbound")
+                            )),
+                            optional("styxHeaders", object(
+                                    optional("styxInfo", object(
+                                            field("name", string()),
+                                            optional("format", string())
+                                    )),
+                                    optional("originId", object(
+                                            field("name", string())
+                                    )),
+                                    optional("requestId", object(
+                                            field("name", string())
+                                    )),
+                                    atLeastOne("styxInfo", "originId", "requestId")
+                            )),
+                            optional("include", string()),
+                            optional("retrypolicy", object(opaque())),
+                            optional("loadBalancing", object(opaque())),
+                            optional("plugins", object(
+                                    optional("active", string()),
+                                    optional("all", map(object(opaque())))
+                            )),
+                            optional("jvmRouteName", string()),
+                            optional("originRestrictionCookie", string()),
+                            optional("responseInfoHeaderFormat", string()),
+                            optional("httpPipeline", object(opaque())),
+                            optional("logFormat", string()),
+                            optional("userDefined", object(opaque())),
+                            optional("requestTracking", bool()),
+                            optional("httpHandlers", map(object(
+                                    optional("name", string()),
+                                    field("type", string()),
+                                    optional("tags", list(string())),
+                                    optional("config", union("type")))))
+                    ));
+    }
+
 
     private ServerConfigSchema() {
     }
 
     static Optional<String> validateServerConfiguration(YamlConfiguration yamlConfiguration) {
         try {
-            STYX_SERVER_CONFIGURATION_SCHEMA.validateObject(yamlConfiguration.root());
+            BUILTIN_HANDLER_SCHEMAS.forEach(STYX_SERVER_CONFIGURATION_SCHEMA_BUILDER::typeExtension);
```

Comment:
Shouldn't `BUILTIN_HANDLER_SCHEMAS.forEach(STYX_SERVER_CONFIGURATION_SCHEMA_BUILDER::typeExtension);` be in the static initialiser instead? So that if we call this method repeatedly (e.g. in tests) we won't repeatedly add the same schemas?